source/ShopPanel.cpp

```diff
@@ -1458,28 +1528,28 @@ vector<ShopPanel::Zone>::const_iterator ShopPanel::Selected() const
 // letter of the button (or ' ' if it's not on a button).
 char ShopPanel::CheckButton(int x, int y)
 {
-	if(x > Screen::Right() - SIDEBAR_WIDTH - 342 && x < Screen::Right() - SIDEBAR_WIDTH - 316 &&
+	const Point click(x, y);
+	const Interface *shopUi = GameData::Interfaces().Get("shop");
+	const Rectangle leaveButton = shopUi->GetBox("leave button");
+	const Rectangle parkButton = shopUi->GetBox("park button");
+	const Rectangle buyButton = shopUi->GetBox("buy button");
+	const Rectangle sellButton = shopUi->GetBox(isOutfitter ? "sell button" : "sell ship button");
+
+	if(leaveButton.Contains(click))
+		return 'l';
+	if(parkButton.Contains(click))
+		return 'k';
+	if(buyButton.Contains(click))
+		return IsAlreadyOwned() ? 'i' : 'b';
+	if(sellButton.Contains(click))
+		return 's';
```

Comment:
Why not just use buttons if you're already moving this to an interface?

Reply:
I haven't figured out how to make their click zones work automatically. If you look in interfaces.txt they ARE buttons, but i could not click them.

Please help if you know what I missed or did wrong.

Reply:
Ah, I guess it needs to be done your way if we want the keypress to be sent even when the button is inactive. But if you have both a button and a rectangle zone, doesn't it trigger twice on a single click?

Reply:
Well if it weren't for the problem that the button definitions don't seem to work, I would have thought the same.
The whole DoButtons() function should be unnecessary but I couldn't get buttons to work here the way they do in other panel Interfaces that use "button".